deprish.py

```diff
@@ -0,0 +1,294 @@
+        associated the common rows' data. We may need different types of
+        reshaping later, so this method is here to accumulate reshaping
+        methods.
+        """
+        if clobber or self.reshaped_df is None:
+            self.reshaped_df = self.split_allocate_common()
+        return self.reshaped_df
+
+    def fill_in(self, clobber=False):
+        """
+        Clean % columns and fill in missing values.
+
+        Args:
+            clobber (bool): if True and dataframe has already been generated,
+                regenergate the datagframe.
+
+        Returns:
+            pandas.DataFrame: depreciation study records that have been cleaned
+            and nulls have been filled in.
+        """
+        if clobber or self.filled_df is None:
+            filled_df = deepcopy(self.reshape(clobber=clobber))
+            # convert % columns - which originally are a combination of whole
+            # numbers of decimals (e.g. 88.2% would either be represented as
+            # 88.2 or .882). Some % columns have boolean columns (ending in
+            # num_or_pct) that we fleshed out to know wether the values were
+            # reported as numbers or %s. There is one column that was easy to
+            # clean by checking whether or not the value is greater than 1.
+            filled_df.loc[filled_df['net_salvage_num_or_pct'],
+                          'net_salvage_pct'] = filled_df.net_salvage_pct / 100
+            filled_df.loc[filled_df['depreciation_annual_num_or_pct'],
+                          'depreciation_annual_pct'] = \
+                filled_df.depreciation_annual_pct / 100
+            filled_df.loc[abs(filled_df.reserve_pct) >= 1,
+                          'reserve_pct'] = filled_df.reserve_pct / 100
+            logger.info(
+                f"# of reserve_pct over 100%: "
+                f"{len(filled_df.loc[abs(filled_df.reserve_pct) >= 1])} "
+                "Higher #s here may indicate an issue with the original data "
+                "or the fill_in method"
+            )
+            # get rid of the bool columns we used to clean % columns
+            filled_df = filled_df.drop(
+                columns=filled_df.filter(like='num_or_pct'))
+
+            # then we need to do the actuall filling in
+            self.filled_df = filled_df.assign(
+                net_salvage_pct=lambda x:
+                    # first clean % v num, then net_salvage/book_value
+                    x.net_salvage_pct.fillna(x.net_removal / x.book_reserve),
+                net_removal=lambda x:
+                    x.net_removal.fillna(x.net_salvage_pct * x.book_reserve),
+                unaccrued_balance=lambda x:
+                    x.unaccrued_balance.fillna(
+                        x.plant_balance_w_common - x.book_reserve
+                        + x.net_removal),
+                reserve_pct=lambda x: x.book_reserve / x.plant_balance_w_common
+            )
```

Comment:
The ordering here does matter in the context of how many values get fleshed out in the end. Because all of these are `fillna`'s instead of brute force calcs, there is no overrides of the existing data.

Mostly `net_removal` needs to be filled in before `unaccrued_balance`.

In your last question, do you mean have I checked whether these calculations could be run across the board and whether we'd get consistent results? Like if we did brute force calculate everything would we get the same results? If so, I have not incorporated that at this point - I did check this a fair amount while doing it in the spreadsheet version of this. but I could incorporate that for sure.